pkgs/tools/text/frawk/default.nix

```diff
@@ -0,0 +1,47 @@
+{ lib
+, rustPlatform
+, fetchCrate
+, libxml2
+, ncurses
+, zlib
+, features ? [ "default" ]
+, llvmPackages_12
+}:
+
+let
+  inherit (builtins) elem;
+in
+
+rustPlatform.buildRustPackage rec {
+  pname = "frawk";
+  version = "0.4.6";
+
+  src = fetchCrate {
+    inherit pname version;
+    sha256 = "sha256-yEdfMikMcsQePxQL1+lma95O1x5z1B7aXAEf8apuGaU=";
+  };
+
+  cargoSha256 = "sha256-osi77Fx8jSfIvAIpThgPbnuJVF/Ydr2/+ROHcDG5ZbA=";
+
+  buildInputs = [ libxml2 ncurses zlib ];
+
+  buildNoDefaultFeatures = true;
+  buildFeatures = features;
+
+  preBuild = lib.optionalString (elem "default" features || elem "llvm_backend" features) ''
+    export LLVM_SYS_120_PREFIX=${llvmPackages_12.llvm.dev}
+  '' + lib.optionalString (elem "default" features || elem "unstable" features) ''
```

Comment:
```suggestion
  preBuild = lib.optionalString (lib.elem "default" features || lib.elem "llvm_backend" features) ''
    export LLVM_SYS_120_PREFIX=${llvmPackages_12.llvm.dev}
  '' + lib.optionalString (lib.elem "default" features || lib.elem "unstable" features) ''
```

Reply:
any reason why `lib.elem` is preferred here? there seems to be more `builtins.elem` then `lib.elem` currently in nixpkgs

```console
$ rg -F "lib.elem " | wc -l
51

$ rg -F "builtins.elem " | wc -l
94
```

Reply:
lib variants of functions are always preferred.

Reply:
I will take your word on this